server/server.go

```diff
@@ -970,3 +970,37 @@ func (s *Server) KillNonFlashbackClusterConn() {
 		s.Kill(id, false)
 	}
 }
+
+// GetMinStartTS implements SessionManager interface.
+func (s *Server) GetMinStartTS(lowerBound uint64) (ts uint64) {
+	// sys processes
+	if s.dom != nil {
+		for _, pi := range s.dom.SysProcTracker().GetSysProcessList() {
+			if thisTS := pi.GetMinStartTS(lowerBound); thisTS > lowerBound && (thisTS < ts || ts == 0) {
+				ts = thisTS
+			}
+		}
+	}
+	// user sessions
+	func() {
+		s.rwlock.RLock()
+		defer s.rwlock.RUnlock()
+		for _, client := range s.clients {
+			if thisTS := client.ctx.ShowProcess().GetMinStartTS(lowerBound); thisTS > lowerBound && (thisTS < ts || ts == 0) {
+				ts = thisTS
+			}
+		}
+	}()
+	// internal sessions
+	func() {
+		s.sessionMapMutex.Lock()
+		defer s.sessionMapMutex.Unlock()
+		analyzeProcID := util.GetAutoAnalyzeProcID(s.ServerID)
```

Comment:
I think it is unnecessary now because the analyze-statement will not be affected by GC after v6.2.0. Maybe it is helpful for v6.1?

Reply:
I think `analyzeProcID` is used to exclude analyze proc [here](https://github.com/pingcap/tidb/pull/39950/files#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4R1000). I just copy the logic of [GetInternalSessionStartTSList](https://github.com/pingcap/tidb/blob/97a9e63d789ea5bace36b82526ea105a1ac65891/server/server.go#L896-L910) here to:
1. let the session manager export a method for calculating the min-start-ts of all sessions (instead of calculating it in `InfoSyncer.ReportMinStartTS`).
2. avoid allocating slices for process info and internal start ts.

Reply:
Ok.